wweval/R/add_horizons.R

```diff
@@ -0,0 +1,73 @@
+#' Add columns indicating the horizon to the quantiles
+#'
+#' @description
+#' This function takes in a tibble of quantiles containing the columns
+#' `date`, `forecast_date` and `period` and adds the following columns:
+#' `horizon_days` (an integer) and `horizon` (a string to be used for
+#' categorical grouping of horizons)
+#'
+#'
+#' @param quantiles A tibble containing the quantiled values and the following
+#' required columns: `date`, `forecast_date`, and `period`
+#'
+#' @return a tibble containing the same columns as `quantiles` plus
+#' `horizon_days` and `horizon`
+#' @export
+add_horizons_to_quantiles <- function(quantiles) {
+  quantiles_w_horizons <- quantiles |>
+    dplyr::mutate(
+      horizon_days = as.numeric(date - forecast_date)
+    ) |>
+    dplyr::mutate(
+      horizon = dplyr::case_when(
+        period == "calibration" ~ "calibration",
+        period != "calibration" & horizon_days < 0 ~ "nowcast",
```

Comment:
if kept as-is, this needs to change to <= 0
```suggestion
        period != "calibration" & horizon_days <= 0 ~ "nowcast",
```